### ISSUE TITLE
Create FG1000 panel variant

### ISSUE THREAD

**Issue by wlbragg** [c172p-team/c172p#1333]
I hope there isn't already an issue for this. I couldn't figure out how to effectively search through the existing issues.

Because glass is now the future of aircraft, I think we need a glass variant. I did the PA-18 and it really wasn't that difficult to retrofit it. We could do the same thing with our flagship and probably should. I think it would have to be done the same way as I did the PA-18 and it would have to have it's own set file and that would be the only way to access it. Not on the fly like the gear and engine changes. It's too heavy and cause the other PA-18 variants to lag even when in the background.

What I think is needed most for this is just some good cockpit reference material the matches our c172p model as best we can find. This winter when I'm holed up maybe I can start on this with a little help here and there.

![maxresdefault](https://user-images.githubusercontent.com/11069680/95544182-0a882f00-09c0-11eb-92a9-453cee4b30af.jpg)

**Comment by wlbragg** [comment]
It's a start!

![yRFvtbX](https://user-images.githubusercontent.com/11069680/95510187-24ebe980-097b-11eb-839f-c2539a780f2d.jpeg)

**Comment by dany93** [comment]
>  is there some changes we would need to make to the 180 to better fit the specs?

From the FG engine point of view, i think there is no difference. A 180 hp engine is a 180 hp engine.
I don't know what are the differences between the actual performances, of the engine with carburetor and the injected engine  (torque vs RPM?) but anyway, the JSBSim engines are simple ones, constant torque if I recall well. 

At my knowledge, an injected engine is much easier to start than a carburetor one. Less capricious.

By elsewhere, the engine lack of feeding at negative g's will have no sense from the carburetor point of view, but it will still exist from the main tanks point of view. Maybe a longer lag?
The carburetor icing will have no sense.
Obviously, no primer.
(I can forget some, I have no experience of real aircraft engine maintenance and performance)

The constant level (collector) tank will have no meaning at all, but that's a (painful...) cheating, artifact for the two engines. Unseen from the user, no effect on the performances.

**Comment by dany93** [comment]
@dany93 wrote
> Obviously, no primer.

I wrote it too quickly and too short.
My reasoning (not based on real life knowledge) was that the engine is already an injected one.

But a priming procedure is written in the [Cessna 172S N552 SP (Engine Model Number: l0-360-L24) POH](http://www.engineersflyingclub.org/planes/552SPPOH.pdf):
(p. 4-12, PDF 96/384)

> lf engine is warm, omit priming procedure of steps 6, 7 and 8 below.
> 
> 6. Auxiliary Fuel Pump Switch -- ON.
> 7. Mixture -- SET to FULL RICH (full forward) until stable fuel
> flow is indicated (usually 3 to 5 seconds), then set to IDLE
> CUTOFF (full aft) position.
> 8. Auxiliary Fuel Pump Switch -- OFF.
> 9. lgnition Switch -- START (release when engine starts).
> 10. Mixture -- ADVANCE smoothly to RICH when engine starts.
> 
> lf engine floods (engine has been primed too much), turn off
> auxiliary fuel pump, place mixture to idle cutoff, open throttle
> 1/2 lo full, and motor (crank) engine. When engine starts,
> set mixture to full rich and close throttle promptly.

**Comment by wlbragg** [comment]
@dany93 thanks for the logical thinking. That gives me a sound starting point.

I've actually had a chance to work on it a bit. I updated the image above to reflect the changes I have made so far.

System wise I temporarily bypassed the primer system if using the glass panel, but I think I will eventually have to create some logic to mimic the fuel pump system.

It looks like the electrical bus is a little different than the current c172p  as there is a bus 1 and a bus 2.

**Comment by tonghuix** [comment]
To be honest, I prefer C172S instead of P variant.

**Comment by wlbragg** [comment]
@tonghuix pictures and information would help, educate me if you have the time! I honestly didn't know there we different versions. If using the 172p for the base then I would lean toward whichever version fit that best.

**Comment by tonghuix** [comment]
@wlbragg I am training in a C172SP with G1000 variant right now.  For the autopilot, it could using either KAP140 or GFC700(which is optional in G1000). If KAP140 is installed, the panel should arrange a place for it; if GFC700 is installed, it would not require dedicate place.

C172 G1000 POH: https://6082a08e-f6ee-43ce-a788-78d22ce0c9ae.filesusr.com/ugd/fe4543_d932b973b08d48519e6e0e69b140adb2.pdf

C172 Training Supplement: https://s3.amazonaws.com/atp-program-docs/supplements/cessna-172-training-supplement.pdf

G1000 Cockpit Ref: https://static.garmincdn.com/pumac/190-00384-12_0A_Web.pdf

G1000 Pilot Guide: https://static.garmincdn.com/pumac/190-00498-07_0A_Web.pdf

**Comment by wlbragg** [comment]
@tonghuix thank you, the data will really help once I study it.

**Comment by wlbragg** [comment]
>To be honest, I prefer C172S instead of P variant.

@tonghuix what is the difference between the two? The Image above of the work I have already done (the work competed that is above the panel that has the ignition and breakers) looks to be the same as in the reference material you provided. So I am wondering why you made the above comment as if I was modeling it after the p and not the s or sp? So far I was just going off a couple pictures I found searching on the web. I thought they were of the s or sp. Can you explain this so I understand your concern, why you may have thought I was modeling the p? Is there a 172p with the G1000? Or, what is the "p" your referring to?

**Comment by tonghuix** [comment]
@wlbragg Well, G1000 is an optional instrument, aircraft owner could install it or not, at least theoretically. However, due to 172P is older one, I do not see a "P" model with G1000.  Another difference is yoke, SP's yoke looks like a little bigger than P's.

I just agree @legoboyvdlp 's opinion, it is better we start modeling 172SP which is a newer model. And it is also a good chance to solve #1310 , make our 172 perfect.

**Comment by wlbragg** [comment]
Ah, OK, I think I understand, thanks!
Currently I am not sure I have the skill set to do much other than convert our p to an sp, which I gather by your explanation is not completely realistic.
If we could get someone to model the sp, that would be ideal.
I might be able to do some of the changes in #1310 depending on what exactly they entail, I was fairly successful in cleaning up the front air intake, wing tips, rudder to tail connection and a bunch of miscellaneous mesh irregularities of the fuselage. I'll take a closer look at some of this and we'll see!

**Comment by legoboyvdlp** [comment]
That's really nice  @wlbragg :) 

I wish the screen could have some anti-aliasing as its very hard to see and the update rate is somewhat low on the MFD. I guess these are for the upstream FG1000 not the 172 though.

**Comment by wlbragg** [comment]
Here is what it looks like at night, so far. Man, do we have a lot off effects to filter through when making a new variant such as this. I still have a lot of work to do on the panel containing the breakers, Throttle and mixture controls need reposition, others need removed. We need some new style stitches and knobs as well.
Procedural lights for the gauges need moved and turned on, other miscellaneous items.
So technically this is a c172p retrofit to a glass panel, but it is being set up in a way that if we get a new c172sp model it should be easy to add this work to it without a while bunch of effort.

![fg1000](https://user-images.githubusercontent.com/11069680/96915918-b408ff00-145b-11eb-84fe-2cbad2fbb217.jpg)

**Comment by wlbragg** [comment]
@legoboyvdlp my anti-aliasing is actually shut off for other reasons, this may look better on a FlightGear installation with it turned on.

**Comment by wlbragg** [comment]
Cleaned off more of the lower panel.

![fgfs-20201024224515](https://user-images.githubusercontent.com/11069680/97093251-29610500-15ff-11eb-8bcf-40b1c1dcf71a.jpg)

**Comment by gilbertohasnofb** [comment]
Wow this is looking really good! Amazing job :clap: 

Just a minor thing I seem to notice in this screen shot: the white avionics switch seems to be rotating on a strange axis, look how the left one does not match the one on the right neither at the top nor at the centre of the switches.

**Comment by tonghuix** [comment]
> 
> 
> Cleaned off more of the lower panel.
> 
> ![fgfs-20201024224515](https://user-images.githubusercontent.com/11069680/97093251-29610500-15ff-11eb-8bcf-40b1c1dcf71a.jpg)

No Audio Panel? It is essential! And you need four nobs to adjust cockpit lights (PFD, MFD, avionics, standby instruments) . Move outside light switches to lefthand under Master and AV switches. 12V port should lower near shut-off valve.

**Comment by tonghuix** [comment]
And once you add audio panels, the autopilot will not a problem, so you can remove KAP-14 now.

**Comment by wlbragg** [comment]
The audio panel is the thin strip between the mfd's?

**Comment by tonghuix** [comment]
> 
> 
> The audio panel is the thin strip between the MFD's?

Yep! Here is a link https://www.faa.gov/training_testing/training/fits/guidance/media/G1000.pdf, it is better have a look at the graph at Pg2, it explain lots of things.

**Comment by wlbragg** [comment]
A little more work done on the panel.

![fg1000](https://user-images.githubusercontent.com/11069680/97235384-be573000-17b0-11eb-8a79-80c0aec3c22f.jpg)

**Comment by wlbragg** [comment]
@tonghuix 
I didn't research what interior lighting goes with what dial or knob. Also, the green light and the Batt Test system, I assume it is the replacement for the Low Voltage light?
If you get a chance to research those systems and feel like taking the time to lay it out for me, I would certainly appreciate it. It might keep me from having to do a bunch of reading and searching for the information.
Also I need to find the bus 1 ans 2 layouts eventually.
Is SW/CB panels the MFD's?

**Comment by wlbragg** [comment]
I'm thinking of giving the choice of using either the KAP140 or the GFC700 when using the FG1000 variant. 
I need to see if I can easily hook up the GFC700 and if it is tuned .at all for the c172

**Comment by wlbragg** [comment]
![GZB0HO9](https://user-images.githubusercontent.com/11069680/97608732-b8479600-19e0-11eb-9247-e7a13db7fd4f.jpeg)

Well, I'm really close to finished with the additional graphics needed for the variant. It's time to add systems and edit systems. The electrical system is probably the biggest issue. Lots of new breakers to tie in.

**Comment by tonghuix** [comment]
Here is a graph for electrical system of C172S: https://solarflyer.files.wordpress.com/2020/01/fuel-elec.pdf
And a electrical system simulator http://avitmedia.aero.und.edu/c172sElectrical/index.html

For the green light, it is only a indicator of STBY BATT, when I turn the switch to TEST position (will spring back to OFF position if no press), it will light up. And any time STBY BATT kicks in, the green light will on as well. For Low Voltage, there should be LOW VOLT annunciation in G1000.

For MFD, AVONIC switch will control it.

**Comment by wlbragg** [comment]
@stuartbuchanan 
I tried adding in the FG1000/GFC700.xml and I am getting a considerable amount of oscillation in the heading axis. Any guide to tuning this would be of great help.

I really like having the option of using the GFC700 built in auto pilot. I do want to keep the ability to choose between the two but that may have to be a variant option instead of a GUI option. I haven't really researched how or if we could do this on the fly.

**Comment by tonghuix** [comment]
> 
> 
> @stuartbuchanan
> I tried adding in the FG1000/GFC700.xml and I am getting a considerable amount of oscillation in the heading axis. Any guide to tuning this would be of great help.
> 
> I really like having the option of using the GFC700 built in auto pilot. I do want to keep the ability to choose between the two but that may have to be a variant option instead of a GUI option. I haven't really researched how or if we could do this on the fly.

Just an info, our school do own some C172Ss equipped with KAP140 autopilot. But most of them are buggy, and not very reliable, mechanics are removing them. A lot of my flight experience with such autopilot are really not good feeling, such us disengage button not work, banking too much, cannot maintain altitude, or twist yoke super sluggishly.

**Comment by wlbragg** [comment]
@stuartbuchanan 
I fixed the GFC700 autopilot. It was either configured twice or competing with the KAP140 so please disregard the tuning help for now

@tonghuix I pushed the changes that use the GFC700 VS the KAP140. The KAP140 still is visible on the panel but is disconnected. If you have the time to test it, that would be great.
Note: breakers and knobs and switches are still not completely configured correctly.

**Comment by wlbragg** [comment]
@tonghuix
You now have a variant choice of either the KAP140 or GFC700. I really like the GFC700, lots of options with that one. I didn't get to test it much though, so it would be helpful if you get a chance to test it.
The amphibious gear control was not correctly placed on the panel for the fg1000 variant so I fixed that as well.
Changed the textures on the breakers to match the specs. I'm now starting on the electrical.

**Comment by tigert** [comment]
> The carburetor icing will make no sense. But I don't know what happens for the air input pipes.
...
> Obviously, no primer. (please see my next message for complement)
> (I can forget some, I have no experience of real aircraft engine maintenance and performance)

The fuel injected version has no carburettor heat, obviously. It has an alternate air intake that opens if the air filter is blocked by ice or something else, that pulls unfiltered air from within the engine cowling. This causes 10% power loss, likely due to warmer air in there.

Found this while I googled for the above details, might be useful too:
https://www.pilotscafe.com/c172s-sp-skyhawk-orientation/

**Comment by tigert** [comment]
> > I really like having the option of using the GFC700 built in auto pilot. I do want to keep the ability to choose between the two but that may have to be a variant option instead of a GUI option. I haven't really researched how or if we could do this on the fly.

> Just an info, our school do own some C172Ss equipped with KAP140 autopilot. But most of them are buggy, and not very reliable, mechanics are removing them. A lot of my flight experience with such autopilot are really not good feeling, such us disengage button not work, banking too much, cannot maintain altitude, or twist yoke super sluggishly.

The GFC700 is a way better integration.

The KAP140 is such a hack anyway, it is not really integrated to the G1000 at all except for roll control (NAV mode), and there is an actual turn coordinator gauge mounted behind the right screen on C172/G1000/KAP140 planes: https://forums.x-plane.org/index.php?/forums/topic/192489-ct182t-xp11-some-observations/&tab=comments#comment-1759568

The KAP140 has its own BARO setting, its own altitude selection, it does not understand any altitudes you might have on the G1000 flight plan. It is just a nightmare in usability. I would rather spend the effort on making a good GFC700 version. :)

The only reason those KAP140 + G1000 Skyhawks exist is because they wanted to get a G1000 plane certified, but Garmin did not yet have its own autopilot ready. And KAP140 was the way they could shoehorn everything in. Early Diamond DA40's have the same thing.

I mean, you can fly it and it works, but just the fact that you have TWO altitude settings on the plane, of which the more obvious one is not working at all except as a fancy postit note, is just ridiculous, and can even be confusing and thus dangerous on a bad day.



Seems like the autopilot servos might wear out

**Comment by stuartbuchanan** [comment]
On Fri, Oct 30, 2020 at 11:57 AM Tuomas Kuosmanen wrote:

> The only reason those KAP140 + G1000 Skyhawks exist is because they wanted
> to get a G1000 plane certified, but Garmin did not yet have its own
> autopilot ready. And KAP140 was the way they could shoehorn everything in.
> Early Diamond DA40's have the same thing.
>
> I mean, you can fly it and it works, but just the fact that you have TWO
> altitude settings on the plane, of which the more obvious one is not
> working at all except as a fancy postit note, is just ridiculous, and can
> even be confusing and thus dangerous on a bad day.
>
> Seems like the autopilot servos might wear out
>
That's just frightening.  I remember reading an early review of the 172
with G1000 and the aircraft reviewer pointed this out.  It may have been
even worse because the standby altimeter also has a separate BARO setting!
So 3 different pressures to change.  Sounds like an airspace
infringement waiting to happen ;)

I agree that simply removing the KAP140 is the way to go.

-Stuart

**Comment by dany93** [comment]
Thanks @tigert for your extra information and your (interesting) link.

**Comment by wlbragg** [comment]
Thanks everyone for the information. It really helps me along the way to get this kind of input.

Well, the option is already available and complete. The additional code is negligible, Two different set files and a conditional model load and that's it.

I can see why the suggestions to leave out the KAP140 though, there is a chance no one will ever use it and when I tried to use it, using the available parts of the MFD panel, I found out how dangerous it is. I actually crashed the aircraft when I entered into a stall and spin at low altitude trying to mess with the altitude for the first time (guessing at how to use it from the GFC700 panel). It is REALLY good practice. I don't have anything else to do to either systems as far as integration, I don't think. Maybe later on someone may want to verify they work together (or not) as in real life. But becasue they were built that way in the beginning, I want to leave the option, as dumb and dangerous as it was and is.

So what's the deal with the images online of the G1000 combined with the KR87? Is there a story behind that?

**Comment by wlbragg** [comment]
>The KAP140 still is visible on the panel but is disconnected.

If anyone was considering this as a factor, this is no longer the way it is. You choose through a variant which version you want and in the GFC700 version the KAP140 is completely gone!

**Comment by wlbragg** [comment]
@tonghuix @tigert 

When you see the following, is it understood that those "essential systems" are powered regardless of avionics1 or avionics2 switch being on? So long as electrical bus1 or electrical bus2 is powered, they have power?

![schm](https://user-images.githubusercontent.com/11069680/97750591-15b21480-1abf-11eb-8702-a8c0219e8dfa.jpg)

**Comment by tonghuix** [comment]
> @tonghuix @tigert
> 
> When you see the following, is it understood that those "essential systems" are powered regardless of avionics1 or avionics2 switch being on? So long as electrical bus1 or electrical bus2 is powered, they have power?
> 
> ![schm](https://user-images.githubusercontent.com/11069680/97750591-15b21480-1abf-11eb-8702-a8c0219e8dfa.jpg)

For best understanding using the simulator to learn the electrical system: http://avitmedia.aero.und.edu/c172sElectrical/index.html

What I understand is there are three electrical buses - Bus 1, 2 and essential bus. If Master switch off, the essential bus is powered by Standby battery (if STBY BATT armed).  When Master switch turn on, the standby battery control unit will kick off the standby battery, so the whole electrial system powered by Main battery or alternator. The essential bus only provide energy to 'essential' devices such us PFD, ADC, AHRS etc. it is designed only kicks in when Main Battery failue.

**Comment by wlbragg** [comment]
@legoboyvdlp @gilbertohasnofb 

I don't know who did the electrical for the c172p but I have a question? I was wondering why we calculate the load or draw as volts? The entire system is 28 volts and the power draw through the breakers is typically amps, isn't it? I also noticed some values that don't make sense to me.
As I am doing a new electrical system for the fg1000 I noticed items like the following for example...
# Flaps, draw = 57
# Pitot Heat Power, draw = 28
# Interior lights, draw = 57
yet the breakers are labeled with amounts such as
# Flaps, draw = 10
# Pitot Heat Power, draw = 5
# Interior lights, draw = 5
So why are we using such a high value? Or do I not understand something fundamental here?

**Comment by tonghuix** [comment]
> @legoboyvdlp @gilbertohasnofb
> 
> I don't know who did the electrical for the c172p but I have a question? I was wondering why we calculate the load or draw as volts? The entire system is 28 volts and the power draw through the breakers is typically amps, isn't it? I also noticed some values that don't make sense to me.
> As I am doing a new electrical system for the fg1000 I noticed items like the following for example...
> Flaps, draw = 57
> Pitot Heat Power, draw = 28
> Interior lights, draw = 57
> yet the breakers are labeled with amounts such as
> Flaps, draw = 10
> Pitot Heat Power, draw = 5
> Interior lights, draw = 5
> So why are we using such a high value? Or do I not understand something fundamental here?

I think it is meared as circle ampere? As my experience, flap and landing/taxi lights are super cost for electrical, so when alternator failure, it draw the Main battery super fast.

**Comment by wlbragg** [comment]
I'm making progress on the electrical, It's slow going as there are a couple technical challenges I'm still trying to overcome. @stuartbuchanan how do you isolate the two fg1000 displays independent of each other? I need to tie one to buss1 and also to essential bus and the other to buss2. I haven't figured out how to do that yet. So far the only method I know of to isolate them work on both, that is the fg1000system.show()/hide() function.

>I think it is meared as circle ampere?

@legoboyvdlp @gilbertohasnofb @tonghuix I'm still not understanding why the buses refer to volts VS amps in the existing code. The items that draw from the bus don't draw volts they draw amps, they are all 28 volt systems as far as I know. And also why or how any device on the system that draws 57 volts (amps) can have a 5 or 10 amp breaker and not trip it instantly? Although I understand there is no mechanism in place to trip breakers.

**Comment by tonghuix** [comment]
The Main Battery is 24 volts, not 28 volts. And also alternator output is 28V with 60A circle. That's why I think it is meared by circle ampere, when flap is operated it cost nearly 57A, close to 60A, so it is reasonble. 

> @legoboyvdlp @gilbertohasnofb @tonghuix I'm still not understanding why the buses refer to volts VS amps in the existing code. The items that draw from the bus don't draw volts they draw amps, they are all 28 volt systems as far as I know. And also why or how any device on the system that draws 57 volts (amps) can have a 5 or 10 amp breaker and not trip it instantly? Although I understand there is no mechanism in place to trip breakers.
> 
> @tonghuix I can;t find the breaker rating of feeder 1 or feeder 2. This is a set of breakers in the cowling that is not available in flight.

Sorry, I really don't know breaker rating...

**Comment by wlbragg** [comment]
>Main Battery is 24 volts

Right, I have it setup that way with the alt producing an additional 3 volts.

>circle ampere, when flap is operated it cost nearly 57A

OK, I guess I have some reading to do. I didn't find a description of "circle ampere"
It must be the draw is 57 amps over the time of flap deployment, but not more than the rated breaker capacity of 10 amps at any point? So for example...
57 amps / 6 second deployment = 9.5 amps draw through the bus for 6 seconds or 57/6=9.5. The flap breaker is a 10 amp breaker.

I've made considerable headway with the fg1000 variant electrical system. It is complex and our current electrical system is missing some key components, like 2 bus+ ess bus + the stb bat system. So recreating the new system from the old system is painstaking. Here is where I am at, the following descriptions are mostly for me as a placeholder.
All buses are created.
stby bat system and bus is mostly created, charges and discharges in arm mode appropriately depending on master bat and alternator setting
Original equipment carryover from the c172p is tied.
Still need to add test lamp illumination on "armed" draw after 5 amp for 10 seconds.
Possibly set up breaker tripping system (failure).
Tie all the fg1000 variant specific equipment to appropriate power bus system,
fuel pump, auto pilot, nav comm, etc.
PFD and MFD is tied to all buses including the stby bat system.

Pushing progress shortly.

**Comment by dany93** [comment]
> I didn't find a description of "circle ampere"

I had the same question. Can it be an approximate translation for 'alternating current' ?. Alternators give a sinusoidal voltage and current. Then, diodes give the 28 V direct voltage (greater than the 24 Volts battery voltage, to charge it).

> 57 amps / 6 second deployment = 9.5 amps draw through the bus for 6 seconds or 57/6 = 9.5. The flap breaker is a 10 amp breaker.

I don't understand either, but it cannot be the explanation. 57 amps burn a 10 amp breaker in less than 6 seconds. And this way of calculating an average is not correct. And if the flaps need 57 amps to deploy, they need 57 amps continually, logically through the flap breaker..

**Comment by dany93** [comment]
I also found this:
(Can no longer be downloaded for free)
An extra battery is installed for G1000 (what for FG1000?)

![Capture du 2020-11-02 11-26-45](https://user-images.githubusercontent.com/11994061/97857685-6a8f9e00-1cfe-11eb-8654-de821bcfc5d5.png)

**Comment by wlbragg** [comment]
@dany93 
I really wish the creator of the original electrical system was available to talk with about the volts VS amps usage in the bus and how/why they use these numbers. It doesn't make sense to me.

>And if the flaps need 57 amps (likely DC?) to deploy, they need 57 amps DC continually during (X) seconds, logically through the flap breaker. Pulled in DC from the battery. With the alternator ON or OFF.

Right, I was just thinking maybe that the 57 amps is the total energy the flaps consumed during their deployment. That they really only need 10 amp to operate but that the creator of the buss charged them with 57 amps over the deployment in total? @tonghuix maybe can describe his knowledge of flap energy usage.

The 57 amps the lights used in the original system, in my system is broke in two. In the original system it was 57 amps for "interior lights", that included the "panel lights". In the new system there is a both a cabin and panel breaker. So I split it and charged each one 28.5 amps. But the breaker for both are 5 amps according to the docs. 
I need a lesson in breakers and amps draw over time I guess.

**Comment by wlbragg** [comment]
I don't know if this is helpful at all but I quickly looked up some common items and their watts usage...

Per hour
Clothes dryer/water heater | 4,000 watts
Water pump | 3,000 watts
Space heater | 1,500 watts
Hair dryer | 1,200 watts

Here is the typical amps and watts for a few devices

Washing machine | 10 | 2200
Toaster | 9.0 | 2000
Tumble dryer | 11.0 | 2500
Dishwasher | 10.0 | 2200

Tumble dryers are typically fed through a 30 amp breaker at 220 volts
The watt usage typically is at and hour of use.

So for flaps
57amps * 24 volt = 1368 watts
or
10 amps * 24 volt = 240 watts

Also these are AC VS DC.

Maybe this is more applicable

12 volt fan
Operating Voltage | 12 Volts DC
Current Rating | 1.5 Amps. & 1.6 Amps.
Power Consumption | 18 Watts & 20 Watts

We know the breaker rating in amps?
We know the voltage is 24?
We should be able to calculate the watts, volts*amps=watts?
We need to know how to draw those watts from a battery over time, the formula?
If I have this information, do I really need anything else?

What power usage labels are on any of this aircraft equipment and how does it read?

**Comment by dany93** [comment]
@wlbragg wrote
> volts*amps=watts?

Yes (P = U*I)

**Comment by legoboyvdlp** [comment]
@wlbragg I'm not sure unfortunately.

**Comment by wlbragg** [comment]
@tonghuix can you verify when the stby batt test light illuminates?
My understanding is...
1) After switching to test if there is a 0.5 amp draw for 10 seconds?
2) If in the arm position once there is a 0.5 amp draw for 10 seconds?

**Comment by wlbragg** [comment]
I pushed what I have so far. @tonghuix if you get a chance to test the PFD, MFD logic with the electrical system using default bus and alternator and the stby batt system.
I still don't have the final or correct test lamp function for the stby batt, but the buss system I think is correct for the items that are hooked up.

**Comment by wlbragg** [comment]
I need pictures of the twin front dome lights and the back passenger dome light, also showing their positions.

**Comment by wlbragg** [comment]
@stuartbuchanan Does the GFC700 logic have any power source logic for something like a breaker? This aircraft has a breaker direct to the autopilot power. I'm looking for that connection if there is one. Otherwise I guess I need to use something like the AP/GA switch?

@tonghuix Can you tell me how that works? If you switch of the breaker, what exactly happens to the AP? Does it show anything in the MFP or does it just take down the mechanical hardware?

If it is hardware only and If the logic isn't there maybe we could simulate it by conditioning whether we can use any of the control filters basses on electricity (breaker)?

**Comment by wlbragg** [comment]
I've done a little investigating and discovered the following.
The load on the buses are wrong in as much as any load that isn't conditioned on a switch as well as the breaker is not correct. Such as...

~~~    # Flaps
    if ( getprop("/controls/circuit-breakers/flaps") ) {
        setprop("/systems/electrical/outputs/flaps", bus_volts);
        load += bus_volts / 57;
    } else {
        setprop("/systems/electrical/outputs/flaps", 0.0);
    }

    # Pitot Heat Power
    if ( getprop("/controls/anti-ice/pitot-heat" ) ) {
        setprop("/systems/electrical/outputs/pitot-heat", bus_volts);
        load += bus_volts / 28;
    } else {
        setprop("/systems/electrical/outputs/pitot-heat", 0.0);
    }
    # Interior lights
    if ( getprop("/controls/circuit-breakers/intlt") ) {
        setprop("/systems/electrical/outputs/instrument-lights", bus_volts);
        setprop("/systems/electrical/outputs/cabin-lights", bus_volts);
        load += bus_volts / 57;
    } else {
        setprop("/systems/electrical/outputs/instrument-lights", 0.0);
        setprop("/systems/electrical/outputs/cabin-lights", 0.0);
    }

    # Gear Select Power
    if ( getprop("/controls/circuit-breakers/gear-select") ) {
        setprop("/systems/electrical/outputs/gear-select", bus_volts);
        load += bus_volts / 5;
    } else {
        setprop("/systems/electrical/outputs/gear-select", 0.0);
    }

    # Gear Advisory Power
    if ( getprop("/controls/circuit-breakers/gear-advisory") ) {
        setprop("/systems/electrical/outputs/gear-advisory", bus_volts);
        load += bus_volts / 2;
    } else {
        setprop("/systems/electrical/outputs/gear-advisory", 0.0);
    }

    # Hydraulic Pump Power
    if ( getprop("/controls/circuit-breakers/hydraulic-pump") ) {
        setprop("/systems/electrical/outputs/hydraulic-pump", bus_volts);
        load += bus_volts / 40;
    } else {
        setprop("/systems/electrical/outputs/hydraulic-pump", 0.0);
    }
~~~

and if the master avi switch is set then a total avi load is calculated regardless of whether or not the device is turned on. Its all thrown at it with...
~~~
load += bus_volts / 20.0;
~~~
So this all need to be fixed in the default electrical as well as I am correcting all of this in the fg1000 electrical system.

**Comment by wlbragg** [comment]
Also, does anyone know anything about this?
~~~
    # Air-cond
    if ( getprop("/controls/circuit-breakers/aircond-pwr") ) {
        setprop("/systems/electrical/outputs/aircond", bus_volts);
        load += bus_volts / 57;
    } else {
        setprop("/systems/electrical/outputs/aircond", 0.0);
    }
~~~
I know we have a blank breaker for this but it isn't being used and needs to be commented out?

**Comment by tonghuix** [comment]
For the Standby Battery switch, when you press to TEST position, the light will illuminate, note that the switch will spring back to OFF position if you don't press.  So for pre-flight procedure, I have to set the switch to ARM position, and light will NOT illuminate. However anytime Standby Battery kicks in, the light will illuminate again. And ampere, I really don't know exactly number. Here is a quota from Pre-Flight procedure, note that why PFD is on because Main Battery not connected (Master switch off at this time), and Standby battery kicks in.

![image](https://user-images.githubusercontent.com/423703/98042378-7b6d0c00-1df1-11eb-99be-ede60aa9da6f.png)
 

For the breakers, there is a breaker designated to GFC700 autopilot, so you maybe need another breaker for the AP.  If you pull the breaker in flight, all connection will disconnected, because no electrical is provided to instruments and mechanical. I never test it in real flight.

**Comment by wlbragg** [comment]
>so you maybe need another breaker for the AP. If you pull the breaker in flight, all connection will disconnected
...
because no electrical is provided to instruments and mechanical. I never test it in real flight.

Right, I'm wanting to know if the PFD autopilot displays show anything when that breaker is pulled or if any annunciator window pops up? Or if the autopilot hardware just stops working?

**Comment by wlbragg** [comment]
Well, the electrical for the c172p is wrong. Needs some work. I'm fixing some of the easy stuff, but there is some more complicated changes that need to be made.
I need a good electrical diagram for it.

**Comment by tonghuix** [comment]
> Right, I'm wanting to know if the PFD autopilot displays show anything when that breaker is pulled or if any annunciator window pops up? Or if the autopilot hardware just stops working?

I never test pull the breaker out in real flight. Well, I found some video for you:

GFC700 training: https://www.youtube.com/watch?v=TnOUxBdM2m4
G1000 Basic: https://www.youtube.com/watch?v=WsXxoqfaTFw
G1000 Garmin Tutorial: https://www.youtube.com/watch?v=I0y6p0ct1m0

**Comment by tonghuix** [comment]
> Well, the electrical for the c172p is wrong. Needs some work. I'm fixing some of the easy stuff, but there is some more complicated changes that need to be made.
> I need a good electrical diagram for it.

Note that, elec system for C172P and C172SP are totally different, here is the  diagram for C172P:

![2020-11-03 21-17-30 的屏幕截图](https://user-images.githubusercontent.com/423703/98061581-285b7f00-1e1a-11eb-8025-a93a27378752.png)

**Comment by wlbragg** [comment]
Thank you for the diagram, I saw this one but didn't have a clear copy of it. 
Looking a little closer, our original electrical is really close to this. Especially once I change the conditionals of when to apply loads to the bus. It'll probably not really be noticeable, but it will be more accurate.

**Comment by tonghuix** [comment]
I pulled the FG1000 branch, but still six-pack instrument, how to enable FG1000?

**Comment by wlbragg** [comment]
You have to select the fg1000 variant at the start from the aircraft selection screen in the launcher.

**Comment by dany93** [comment]
I don't understand the logic behind the 'load' in @wlbragg's nasal code in [this message](https://github.com/c172p-team/c172p/issues/1333#issuecomment-721302996).

When I read 
`load += bus_volts / 57`,
 what does 'load' means?

My preference (hypothetical guess) would be that 'load' is the current pulled by a system. Each time you activate one more system, its current is added to the previous total load from the others.

But this is not consistent with the nasal code such I understand (??) it.

From the well known electrical formula `P = U*I`, i.e. Power (W) = Voltage (V) * Intensity (A),
`I = P / U`
Not consistent with load = bus_volts / 57 for one system as written in the nasal code, if 'load' is the current.
Notice that, to calculate the intensity (= load?) due to a system, you have to know its consumed power (in Watts).
Could it be that 57, 5, 40, 20, etc... would the pulled power, with mistakes in the formulas?

Either something is wrong, or I understand nothing the logic and the nasal code (I'm not very good in nasal).

**Comment by wlbragg** [comment]
OK, this is the battery
~~~
BatteryClass.new = func (x) {
    var obj = { parents : [BatteryClass],
      ideal_volts : 24.0,
      ideal_amps : 30.0,
      amp_hours : 3.1875,
      charge_percent : getprop("/systems/electrical/battery-charge-percent/"~x) or 1.0,
      charge_amps : 7.0 };
    setprop("/systems/electrical/battery-charge-percent/"~x, obj.charge_percent);
    return obj;
}
~~~
Note: the charge_percent, this is how this system keeps track of the available power in the battery. It is a percentage of ?.

This is the volts output function of the battery. it uses the charge_percent of the battery
~~~
BatteryClass.get_output_volts = func {
    var x = 1.0 - me.charge_percent;
    var tmp = -(3.0 * x - 1.0);
    var factor = (tmp*tmp*tmp*tmp*tmp + 32) / 32;
    return me.ideal_volts * factor;
}
~~~

**Comment by wlbragg** [comment]
All I have really done so far is...
1)  make the new buses conform to this existing system.
2) correct whether or not any single device tied to any single breaker is actually drawing power or not off this system at any given point in time.
3) added another battery to this system including the logic as to when it is in use. Note the new stby batt is using the same class as the main bat and that can't be right because i think it is a smaller battery with far less capacity. I would need to rewrite the class and add the ability to pass the initial, ideal_volts, ideal_amps and amp_hours. I already modified the class to track individual batteries independently. That was the minimum that had to be done.

**Comment by wlbragg** [comment]
So if we can come up with...

1) the correct formulas for drawing and charging
2) the correct values for the stby battery of ideal_volts, ideal_amps and amp_hours.

I can easily modify this this.

**Comment by wlbragg** [comment]
@dany93  to answer this question
>I don't understand the logic behind the 'load' in nasal code in this message from @wlbragg.

For example, this is how the flaps were configured...
~~~
    if ( getprop("/controls/circuit-breakers/flaps") ) {
        setprop("/systems/electrical/outputs/flaps", bus_volts);
        load += bus_volts / 57;
    } else {
        setprop("/systems/electrical/outputs/flaps", 0.0);
    }
~~~
So effectively, if the flaps breaker was closed and the bus was receiving power, then the draw on the battery was  bus_volts / 57 at all times. It should also be conditioned on whether or not the flaps motor was actually in use.
This one was a bit tough to figure out becasue flaps deployment uses a 4 step normalized value
Flaps can be any of the following 0, .3333, .6666, 1. But when in operation between those values it is anything between 0 and 1 but not any of those 4 values. I couldn't figure out any available property to monitor directly for the value that captures movement. So I wrote my own and included it in the breaker statement as follows.
~~~
var old_flap_position = 0;
var current_flap_position = getprop("/surface-positions/flap-pos-norm");
...
    # Flaps 10 amp breaker
    if (getprop("/controls/circuit-breakers/flaps")) {
        setprop("/systems/electrical/outputs/flaps", bus_volts);
    } else {
        setprop("/systems/electrical/outputs/flaps", 0.0);
    }
    current_flap_position = getprop("/surface-positions/flap-pos-norm");
    if (current_flap_position != old_flap_position) {
        old_flap_position = current_flap_position;
        if (getprop("/systems/electrical/outputs/flaps") > 12) {
            load += bus_volts / 57;
            settimer(func(){
                load += bus_volts;
            }, 3);
        }
    }
~~~
I did that for the amphibious landing gear as well.
There were other items like the avionics equipment that were drawing power even when they were off.
All that logic is straitened out in both the default and fg1000 electrical systems, but not necessarily the correct values.

Now what I haven't done is check the new draw totals against any instruments (volts meter). That is something that needs testing, at some point, by those used to looking at and understanding these numbers.

**Comment by wlbragg** [comment]
Oh, and I must say, I have no idea how accurate our total drain on the battery is over time. I can tell you it still seems quit fast.
If someone has time it would be interesting to test how long it takes to drain the battery with a small load of X value and compare it to real life known values. Maybe that all gets corrected if we figure out the correct formula for drawing down the battery.

I've still have a ton to do, audio panel, finish the stby batt test lamp illumination logic, fuel pump logic (in whatever form, ie: simple or complex), and other stuff I can't remember at the moment.

The most time consuming for me I think will be to learn and understand how all the avionics tie to the fg1000. Those connections may not be there or may require modification. Things like the old com-nav were on the same radio and used the same bus connection. Now it is is separate bus connections and how do I trip, say the nav breaker, and have that effect only the nav portion of the fg1000? Maybe @stuartbuchanan  can shed some light on that portion.

**Comment by wlbragg** [comment]
OK, enough for everyone to filter through and see where I am at and how you all might be able to help sort some of this stuff out if you know or can find out.
Find me the info and I will gladly put in the time required to do the work. The hard part is identifying and then conveying the issues that need researched and then how to resolve them.

**Comment by tonghuix** [comment]
I tested, the elec system is fantastic! But for G1000 it self, a lot of work should do, I think we should programing at FG1000 base code.

**Comment by wlbragg** [comment]
Absolutely, that is where I'm at. I think the base fg1000 code needs some hooks added to it. Although, I really haven't had the chance to learn how it all works, what does and what doesn't. When you come across issues with the fg1000 and how it interfaces to any of the equipment, please post it here so we have a list of issues to sort out.

**Comment by wlbragg** [comment]
Pushed some audio panel additions. I have all the buttons push-able with their indicator light illuminating.
Created logic for...
Com1 Mic
Com2 Mic
Com1
Com2
MKR Mute
DME
ADF
Nav1
Nav2
I didn't really test anything except for the Com's logic so please let me know what is wrong and what the desired outcome should be when you find mistakes.

**Comment by dany93** [comment]
@dany93 wrote on [04 Nov 2020](https://github.com/c172p-team/c172p/issues/1333#issuecomment-721670678)

> Either something is wrong, or I understand nothing in the logic and the nasal code (I'm not very good in nasal).
load += bus_volts / 57,
Could it be that 57, 5, 40, 20, etc... would the pulled power, with mistakes in the formulas?

Now and with this, I understand that 57, 5, 40, 20, etc... is akin to a resistance, which gives a current (load) I = V / R.

**Comment by wlbragg** [comment]
Ah, OK, I'll take a look at that formula and see what its all about. I haven't looked yet but I am interested in getting a correct load reading in the PFD. I briefly read about where it was located in the PDF, but I don't know yet exactly what it is displaying, and definitely not the formula we need to get the value for the display. Maybe I = V / R is the correct formula.

**Comment by wlbragg** [comment]
>Pushed some audio panel additions. I have all the buttons push-able with their indicator light illuminating.
Created logic for...
Com1 Mic
...
I didn't really test anything except for the Com's logic so please let me know what is wrong and what the desired outcome should be when you find mistakes.

I assume the Com'x Mic portion of the Audio Panel really doesn't do anything in the simulator as we don't have any transmission capability in the sim?

So I simply insure the com radio is turned on, turn the volume to 100, and then also activate the Com (receive) button to the right per the documentation. If you only select the Com (receive) button (right) then I also turn on the com radio and turn the volume to 100. 

If anyone knows I'm wondering exactly how the comm radios work in the fg1000, are they always on?
Does the Com button (rec, right button) just connect the radio to the speaker or headset and the volume is left at what? You have a volume knob on the PFD, does it retain the individual volumes for com1, com2, nav1, nav2, marker? Or is it the same volume for each? So a Com radio could be turned down and when you hit the Com (rec) button you still might not hear it becasue it is turned down on the PFD?

**Comment by wlbragg** [comment]
I discovered the FG1000 source comes with a audio panel (GMA1347). I looked for it initially but failed to see it so I built my own. The one Stuart made was only the base template and no ties to the actual equipment (as far as I noticed). So I decided to use the one I built because I had already done most of the work to tie it into the equipment and also the physical buttons are animated. None of the PFD, MFD or audio panel buttons on the fgdata supplied models has animated buttons. I would prefer to eventually animate those buttons and knobs, but I guess there was a reason Stuart didn't do that already?
What a great bit of work by @stuartbuchanan though, hats off.

**Comment by wlbragg** [comment]
I'm looking at the 182 fuel pump and fuel system. It is going to be a big job to do it right and will require isolation same as the electrical was isolated.

**Comment by wlbragg** [comment]
FYI: A bug crept into non-fg1000 variants electrical system. I just fixed it and pushed the fix. The default electrical now tracks the load for all equipment. Total draw if everything is running can be as high as 320, the breaker trips at 330.
I also fixed the electrical logic to track radio power and radio lighting  independently. As I read the docs, the lighting on the radio is controlled by the radio lighting rheostat which is on a different breaker than the radio power.

The Issue-1307 (compositor branch) is paralleling this work for anyone that is using the compositor.

There is an issue with the fg1000 EIS logic. I don't think @stuartbuchanan has done any more work on making it easy to localize custom EIS logic. I haven't completely finished researching what can be done to customize the EIS system. Right now it looks like we have to have any customizing files located in fgdata becasue of some of the core logic's path statement not having enough flexibility to call data from more than one location. But I think if it is accepted I could create a stand alone set of files that could be pushed to fgdata for the c172p. The EIS data that is problematic is the engine[0] VS engine[1] or active-engine. and now the customizable fuel system tank[0]tank[1] VS tank[2]/tank[3]. I was successful with custom files working in fgdata for the engine portion but not yet the fuel portion. Trying to change the class names for those files didn't work right out of the box, or I missed something.

@stuartbuchanan maybe you have a better solution or maybe I can help you with optimising the EIS code to alow for better customization. As I see it now I don't think you could "easily" support even a duel engine aircraft or an aircraft with multiple (more than 2) tanks.

**Comment by wlbragg** [comment]
If you want the EIS to work correctly, for now, when you download the branch you'll find four files in Nasal/Interfaces
SelectableEISPublisher.nas
SelectableFuelInterface.nas
SelectableFuelPublisher.nas
SelectableInterfaceController.nas

You will need to copy the files into the fgdata/Aircraft/Instruments-3d/FG1000/Nasal/Interfaces/ folder.
The only thing I am still unsure of is exactly what the "Fuel QTY GAL" display on the EIS is actually supposed to be displaying. Is it actual gallons in the tank? Why then can you change it using the SYSTEM/RST FUEL/GAL REM/+/- GAL's?

**Comment by wlbragg** [comment]
@tonghuix the POH describes the starting procedure for cold and warm engine. But it doesn't describe how hard it is to start the engine without following the POH guidelines?
What I am asking is, if you don't follow the guidelines, will the aircraft still start?
Do you have to turn on the fuel pump when cold?
If you turn it on when warm, will it always  or likely flood the engine?
In order to program this realistically I need to know the off nominal behavior of this system and what exactly causes it?

**Comment by stuartbuchanan** [comment]
On Tue, Nov 10, 2020 at 1:40 AM wlbragg wrote:

> If you want the EIS to work correctly, for now, when you download the
> branch you'll find four files in Nasal/Interfaces
> SelectableEISPublisher.nas
> SelectableFuelInterface.nas
> SelectableFuelPublisher.nas
> SelectableInterfaceController.nas
>
> You will need to copy the files into the
> fgdata/Aircraft/Instruments-3d/FG1000/Nasal/Interfaces/ folder.
>
Do you need a hand fixing this bit?

> The only thing I am still unsure of is exactly what the "Fuel QTY GAL"
> display on the EIS is actually supposed to be displaying. Is it actual
> gallons in the tank? Why then can you change it using the SYSTEM/RST
> FUEL/GAL REM/+/- GAL's?
>
My understanding is that while there are good fuel flow sensors in the 182
(and presumably the 172) there isn't an accurate fuel level measurement.
So you have to manually update the fuel quantity when you refuel, and then
the fuel flow sensors will accurately reduce the quantity remaining during
the flight.  RST FUEL should set it back to full fuel.

-Stuart


>

**Comment by wlbragg** [comment]
@stuartbuchanan 

>The LEAN and SYSTEM selectors look like they should be changing the EIS display?
![display](https://user-images.githubusercontent.com/11069680/98703489-27a67980-2341-11eb-9bf9-0a9accde80af.jpg)

I think this data is missing from the fg1000 EIS for the c172p? The documentation was referring to this data and I think I confused it with the FUEL QTY Gal indicator graphic and needles.

>Do you need a hand fixing this bit?

Absolutely, I think those files I pushed are good to go into fg1000 in fgdata, unless you have a better way to handle it.

>there isn't an accurate fuel level measurement.
So you have to manually update the fuel quantity when you refuel, and then
the fuel flow sensors will accurately reduce the quantity remaining during
the flight.  RST FUEL should set it back to full fuel.

Ok, makes sense. Again, my confusion was the displays above showing the outcome of those calculations in the EIS.

If there is anything I can do to help with any of this, let me know.

The next thing I have to do regarding the fg1000 is the amp draw. I don't think it is reading and rendering that information correctly, I need to verify this and determine why.

**Comment by tonghuix** [comment]
> @tonghuix the POH describes the starting procedure for cold and warm engine. But it doesn't describe how hard it is to start the engine without following the POH guidelines?
> What I am asking is, if you don't follow the guidelines, will the aircraft still start?
> Do you have to turn on the fuel pump when cold?
> If you turn it on when warm, will it always or likely flood the engine?
> In order to program this realistically I need to know the off nominal behavior of this system and what exactly causes it?

Sadly, I never doing off normal behavior of this system. But for cold weather, my experience is the engine will not easily start when the oil temperature below 10C, so need some warm before start. For hot weather or just restart engine very short priod after shutdown (oil temperature above 150F), due to vapor lock in induction system pipes, it is easily flood the engine, so prime 3 sec is enough, and add throttle 1/2 inch in or half of the full range.

**Comment by D-ECHO** [comment]
Hi there,
Great work!

Not sure whether you're already aware of this, but the autostart menu options on this variant fails with the log message
`   76.49 [ALRT]:nasal      Nasal runtime error: No such member: reset_battery_and_circuit_breakers
   76.49 [ALRT]:nasal        at /home/delta/Aircraft/c172p/Nasal/c172p.nas, line 13
   76.49 [ALRT]:nasal        called from: /sim/bindings/menu/binding[223], line 1`
(originally reported by User Newo on the FG discord)

**Comment by NewoIsTaken** [comment]
Also, it seems like when I arm the standby battery, the PFD doesn't turn on.

**Comment by hbeni** [comment]
> Sadly, I never doing off normal behavior of this system. But for cold weather, my experience is the engine will not easily start when the oil temperature below 10C, so need some warm before start. For hot weather or just restart engine very short priod after shutdown (oil temperature above 150F), due to vapor lock in induction system pipes, it is easily flood the engine, so prime 3 sec is enough, and add throttle 1/2 inch in or half of the full range.

Hi @wlbragg,
I'm not sure if you already looked at my injection simulation at the c182s/t, but the temperature sensitive start (and also warm start without priming needs) is already simulated there. I did plenty of research for the numbers, but didn't found anything concrete; so i tried to model it by gut feeling i got from lots of articles and videos. i think it is relatively good. At about 0° it gets notably harder to prime correctly and it will get really hard to prime when going to about -10°C or lower, and impossible at -25°C (technically, but for humans this is not possible to achieve). You definitely should use the preheater in those conditions (< -6°C the POH says its a must).
The code for all that is mostly in `fuel.xml`.

For the EIS display, i already have tried to get that working, but i really think there is something missing in the FG1000 core (or i just lack the knowledge to interpret the nasal):
- Need CHT in the EIS: HHS81/c182s/issues/354
- Need to display battery amps: HHS81/c182s/issues/355

**Comment by wlbragg** [comment]
@hbeni I did look at it and my first reaction was one of intimidation. But I will get back to it and I think it is the way I want to go, I am also intimidated about making my own so I think in the long run adopting yours would definitely be the way to go. It's just a bit more complicated than I was expecting, nothing about what you did, only my own comfort in my abilities to adopt and implement it.

**Comment by wlbragg** [comment]
@D-ECHO 
I can't reproduce the autostart nasal error, was that maybe after a restart or something else?

@NewoIsTaken
Also I can't reproduce this, If I have everything off and throw that switch the

**Comment by wlbragg** [comment]
Wow, don't know how that happened, accidentally closed the issue there for a time?

**Comment by tigert** [comment]
> I'm thinking of giving the choice of using either the KAP140 or the GFC700 when using the FG1000 variant.
> I need to see if I can easily hook up the GFC700 and if it is tuned .at all for the c172

Ugh, the KAP140 exists in early G1000 C172P's only because the GFC700 was not yet certified at the time. The integration is horrible, the Garmin only drives it left-right and you need to set altitude, baro, vertical speed etc modes in the KAP140. There is actually a single turn&bank gauge *behind the right screen (MFD) of the G1000* that is the attitude / heading reference for the KAP140, the G1000 just says "turn left, now turn right, now straight". Ugh,

The G1000 has all that functionality but it cannot be used (and like the alt selector "works" in the G1000 but it does absolutely nothing apart from being a "postit note" so you can remember the number if you dial it in.

I mean, by all means, give a choice but I don't understand who would be such a masochist to choose it if GFC700 is an alternative... :-)

**Comment by stuartbuchanan** [comment]
I'll take a look at the CHT and battery amps.  The problem is almost
certainly that the wrong properties are being referenced in the EFIS.nas

RE: KAP140 - I suspect you'd struggle to get accurate integration between
the KAP140 and the FG1000.  You'd probably need to add an Emesary node to
listen to the correct messages from the UI and then pass them onto the
KAP140.  I really don't think it's worth the effort.

-Stuart

**Comment by wlbragg** [comment]
>Just an info, our school do own some C172Ss equipped with KAP140 autopilot. But most of them are buggy, and not very reliable, mechanics are removing them

That is why I choose to leave the choice. Because of statements like "our school do own some C172Ss equipped with KAP140 autopilot". I think of it the same I do about any upgrades done to any aircraft. We are always improving on systems and technology in the real world, but that doesn't erase the history of what used to be, I really think it is important to retain that history when possible and practical. If I hadn't evolved this variant in the steps I took, which was using the KAP140 first, then adopting the GFC700, just like the timeline in real life, I may not have given this choice. But once the work is done, I see no need to remove it. It is a reality that actually existed and I guess I leave it mostly for nostalgia.

**Comment by wlbragg** [comment]
>RE: KAP140 - I suspect you'd struggle to get accurate integration between
the KAP140 and the FG1000. 

@stuartbuchanan 
I think @tigert is referring to the implementation of the GFC700 in the "real world"G1000 equipped aircraft. And yes the integration with the fg1000 is about the same, so I leave it alone. It is a variant and has no connection anymore with the fg1000/GFC700 variant, no more work to be done. It's a done deal.

**Comment by wlbragg** [comment]
>I'll take a look at the CHT......
.  The problem is almost certainly that the wrong properties are being referenced in the EFIS.nas

@stuartbuchanan 
I think it is more than that, I think you may not have ever implemented the "Fuel Calc" portion.
See https://github.com/c172p-team/c172p/issues/1333#issuecomment-724826960

**Comment by hbeni** [comment]
What i don't really understand - why does the G1000 do its own calculations on fuel, if it could just use the output of the tanks sensors?

**Comment by stuartbuchanan** [comment]
@wlbragg - Yes, you're right.  I was getting confused between the EGT and
CHT gauges.

On Thu, Dec 17, 2020 at 9:08 PM Benedikt Hallinger wrote:

> What i don't really understand - why does the G1000 do its own
> calculations on fuel, if it could just use the output of the tanks sensors?
>
I think it's because IRL the fuel flow sensors are far more accurate than
the fuel tank sensors.

BTW - I've fixed the issue that meant you have to put files into fgdata.
I'll email you a patch.

**Comment by NewoIsTaken** [comment]
@wlbragg This is what is happening: [video](https://litecomet.com/problem.mp4).

**Comment by wlbragg** [comment]
@NewoIsTaken @D-ECHO 
I haven't had much time to take a real close look at this but I will soon. However, are you sure you are using the most current version on the repository. Because I so far cannot reproduce this. I switch to ARM and the PFD lights up. Also, I vaguely remember actually seeing that nasal error in the past and I thought I got rid of it. But I also don't remember if it had to do with a reset or something like switching airports or seaports on the fly?

**Comment by NewoIsTaken** [comment]
@wlbragg No problem, just tell us if you uncover something!

**Comment by D-ECHO** [comment]
Unfortunately, this still happens with newest FG and c172

**Comment by wlbragg** [comment]
@D-ECHO  @NewoIsTaken I just pushed a small change to the interface controller code via a patch from Stuart. Please pull a fresh copy, make sure you clear your aircraft data, and see if your current issue is fixed. If not, make sure you send me the exact error. Because I am getting no errors and the backup bat logic is working correctly for me.

**Comment by NewoIsTaken** [comment]
@wlbragg, PFD comes on right as I hit the standby battery on switch now!

**Comment by wlbragg** [comment]
@NewoIsTaken  Thanks!

**Comment by wlbragg** [comment]
@NewoIsTaken @D-ECHO 

I pushed some new interior lighting logic for the panel knobs. All that is left for the interior lighting is the Pedestal light which will be a new lightmap.
Also I may still add some low spec rendering lighting controllable with the knobs
The audio panel legend should be lit up as well, controllable with the Avionics knob, but isn't working, a bug I haven't tracked down yet.

I'm, not sure how much the electrical has changed from my last push, only that I am pretty happy with it at this point.

**Comment by NewoIsTaken** [comment]
> @NewoIsTaken @D-ECHO
> 
> I pushed some new interior lighting logic for the panel knobs. All that is left for the interior lighting is the Pedestal light which will be a new lightmap.
> Also I may still add some low spec rendering lighting controllable with the knobs
> The audio panel legend should be lit up as well, controllable with the Avionics knob, but isn't working, a bug I haven't tracked down yet.
> 
> I'm, not sure how much the electrical has changed from my last push, only that I am pretty happy with it at this point.

Great! Thanks!

**Comment by wlbragg** [comment]
Notes:
tie 
controls/lighting/pedistal-norm
controls/lighting/swcb-norm
controls/lighting/knob-avionics
controls/lighting/radio-norm is already tied 
to ra-radiance material (default renderer) and als-lights.

move ELT slightly to fit around amphibious gear control

fix broken transponder select

**Comment by NewoIsTaken** [comment]
@wlbragg I will try to test out the FG1000 today or sometime soon.

**Comment by wlbragg** [comment]
Notes to myself:
save/load
flight recorder
tutorials
add breakers to avionics?
more c172p / fg1000 isolation?

**Comment by wlbragg** [comment]
OK, I pushed a major refactoring of the c172sp electrical. I think it is in really good shape. It includes stby amp and volt readings for the EIC display once I get the changes pushed to fgdata. Using the formula amps * bus volts for each power supply circuit and then converting the total watts back to amps using watts / bus volts corrects the electrical power usage in electrical.nas and electrical-fg1000.nas. So it is easy to assign the actual draw per instrument. If anyone has the time to really look at the numbers there are some debug properties that can be un-commented out in order to watch what's going on in real time.
I think I am to the point of finishing up a few miscellaneous items such as save/load parameter, flight recorder updates and maybe some fg1000 tutorials. Please check it out and test where you can, everyone.

**Comment by wlbragg** [comment]
Another major refactoring pushed. Also cleaned up flight recorder, MP, hot spot text. The major refactoring was to isolate c172p exclusive content from c172sp exclusive content. It should lighten the load slightly for all variants and helps group things in a more understandable fashion.
I think we are really close to being able to make a 2020.4 version tag and see if we can get @stuartbuchanan to put it in fgdata next so we can get some serious testing.
So @legoboyvdlp and @dany93 and any others that can make some time to test this latest refactoring, it would be greatly appreciated. Barring any major issues, I think maybe in a week or two I will ask for a merge and then create a new post compositor release.
